feature_engine/transformation/log.py

```diff
@@ -164,3 +164,206 @@ def _more_tags(self):
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+
+        # calculate C to add to each variable
+        if self.C == "auto":
+            self.C_ = X[self.variables_].min(axis=0).abs() + 1
+        else:
+            self.C_ = self.C
+
+        # check contains zero or negative values
+        if (X[self.variables_] + self.C_ <= 0).any().any():
+            raise ValueError(
+                "Some variables contain zero or negative values after suming"
+                + "constant C, can't apply log"
+            )
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the variables using log transformation.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame
+        ValueError
+            - If the variable(s) contain null values
+            - If the df has different number of features than the df used in fit()
+            - If some variables contains zero or negative values after suming constant C
+
+        Returns
+        -------
+        X: pandas dataframe
+            The dataframe with the transformed variables.
+        """
+
+        # check input dataframe and if class was fitted
+        X = super().transform(X)
+
+        # check contains zero or negative values
+        if (X[self.variables_] + self.C_ <= 0).any().any():
+            raise ValueError(
+                "Some variables contain zero or negative values after suming"
+                + "constant C, can't apply log"
+            )
+
+        # transform
+        if self.base == "e":
+            X.loc[:, self.variables_] = np.log(X.loc[:, self.variables_] + self.C_)
+        elif self.base == "10":
+            X.loc[:, self.variables_] = np.log10(X.loc[:, self.variables_] + self.C_)
+
+        return X
+
+    def inverse_transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the variables using exponential transformation.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be detransformed.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame
+        ValueError
+            - If the variable(s) contain null values
+            - If X has different number of features than X used in fit()
+            - If some variables contains zero or negative values
+
+        Returns
+        -------
+        X: Pandas dataframe
+            The dataframe with the transformed variables.
+        """
+
+        # check input dataframe and if class was fitted
+        X = super().transform(X)
+
+        # check contains zero or negative values
+        if (X[self.variables_] <= 0).any().any():
```

Comment:
this check is suitable for transform, but not for inverse_transform, values can be negative.

This was probably in the PR i linked before, but we fixed it there already and we made a new PR with the updates. Sorry